packages/x-charts/src/ChartsLegend/LegendPerItem.tsx

```diff
@@ -68,6 +70,18 @@ export interface LegendPerItemProps
    * @default 10
    */
   padding?: number | Partial<CardinalDirections<number>>;
+  /**
+   * Callback fired when a legend item is clicked.
+   * @param {React.MouseEvent<SVGRectElement, MouseEvent>} event The click event.
+   * @param {LegendItemParams} legendItem The legend item data.
+   * @param {number} index The index of the clicked legend item.
+   * @default undefined
+   */
+  onItemClick?: (
+    event: React.MouseEvent<SVGRectElement, MouseEvent>,
+    legendItem: LegendItemParams,
+    index: number,
```

Comment:
`LegendItemParams` has been created for display purpose. It contains only the information to display an element. To react to a click user will need more information.

For now it returns
``
{ color, label, id }
``

As a user if I want to react to the click, I want to get information about the series, or the category. I don't care about the color, or the label, and not sure about what is in the `id`

I would expect something more like

``
{ seriesId } // Plus some additional data if you want
``

and
``
{ colorMapId, categoryIndex, minValue, maxValue }
``

Reply:
Didn't see what to add to `colorMapId, categoryIndex`. 

I've improved the typings for when using this in color vs series mode. 

Pie charts are a bit of an outlier. I would probably like to try to make pie chart series closer to regular chart series in V8 😅 

I've also added an interactive demo so users can see the props when they click.